gen2-rgbd-projection/main.py

```diff
@@ -0,0 +1,330 @@
+                        pcl_frames[i], (res["width"], res["height"]), cv2.INTER_CUBIC)
+                else:
+                    depth_frame = np.array(image.getFrame())
+                    if align_depth:
+                        if not align_on_host:
+                            pcl_frames[i] = cv2.resize(
+                                depth_frame, (res["width"], res["height"]), cv2.INTER_NEAREST)
+                        else:
+                            # Approach 1: Use opencv's register depth method
+                            depth_scale = 10.0  # extrinsic is in cm, but depth in mm so convert
+                            depth_dilation = True
+                            scaled_depth_frame = depth_frame.astype(
+                                float) / depth_scale
+                            pcl_frames[i] = cv2.rgbd.registerDepth(
+                                rectified_right_intrinsic, rgb_intrinsic, rgb_distortion, right_to_rgb_extrinsic, scaled_depth_frame, (res["width"], res["height"]), depth_dilation)
+                            pcl_frames[i] = np.array(
+                                pcl_frames[i] * depth_scale).astype(np.uint16)  # convert back from cm to mm
+                            # Approach 2: Use slow but easy to interpert function
+                            # pcl_frames[i] = alignDepthToRGB(depth_frame, pixel_coords, inverse_rectified_right_intrinsic, rgb_intrinsic, right_to_rgb_extrinsic, res["width"], res["height"])
+                    else:
+                        pcl_frames[i] = depth_frame
+
+            if any([frame is None for frame in pcl_frames]) and align_depth:
+                print("Error: Need rgb AND depth frame to align!")
+                continue
+            elif pcl_frames[0] is None:
+                print("Waiting on depth image to visualize")
+                continue
+
+            if align_depth:
+                # Convert nonzero depth into red pixels in 3-channel image
+                depth_three_channel = np.zeros_like(pcl_frames[1])
+                depth_three_channel[:, :, 2] = (
+                    255 * pcl_frames[0].astype(float) / max_disparity).astype(np.uint8)
+                cond = depth_three_channel[:, :, 2] > 0
+                depth_three_channel[cond, 2] = 255
+                # Blend aligned depth + rgb image
+                blended_image = 0.6 * depth_three_channel.astype(float) / 255 + \
+                    0.4 * pcl_frames[1].astype(float) / 255
+                blended_image = (255 * blended_image.astype(float) /
+                                 blended_image.max()).astype(np.uint8)
+                cv2.imshow("RGBD Alignment", blended_image)
+                if not no_pcl:
+                    pcl_converter.rgbd_to_projection(
+                        *pcl_frames, downsample=downsample_pcl)
+                    pcl_converter.visualize_pcd()
+            else:
+                quantized_depth_image = quantizeDisparityFrame(
```

Comment:
`quantizeDisparityFrame` Change it to depth. We are not using disparity anywhere in this code.

Reply:
Correct, typo on my part. That should be depth. Fixed :+1: